ocaml/xapi-aux/throttle.ml

```diff
@@ -39,3 +39,27 @@ module Make (Size : SIZE) = struct
 
   let execute f = execute (get_semaphore ()) f
 end
+
+module Batching = struct
+  type t = {delay_before: Mtime.span; delay_between: Mtime.span}
+
+  let make ~delay_before ~delay_between = {delay_before; delay_between}
+
+  (** [perform_delay delay] calls {!val:Thread.delay} when [delay] is non-zero.
+
+    Thread.delay 0 provides no fairness guarantees, the current thread may actually be the one that gets the global lock again.
+    Instead {!val:Thread.yield} could be used, which does provide fairness guarantees, but it may also introduce large latencies
+    when there are lots of threads waiting for the OCaml runtime lock.
+   *)
+  let perform_delay delay =
+    if Mtime.Span.is_longer delay ~than:Mtime.Span.min_span then
```

Comment:
note: `min_span` is `zero`, they cannot be negative.

Reply:
Changed it to `Mtime.Span.zero` to be more explicit (the intention is to check for >0) delays.